src/environmentd/src/http/webhook.rs

```diff
@@ -55,6 +61,57 @@ pub async fn handle_webhook(
     Ok(())
 }
 
+async fn validate_request(
+    validate_expr: AppendWebhookValidation,
+    body: Bytes,
+    headers: &http::HeaderMap,
+) -> Result<(), WebhookError> {
+    // Convert our headers into a map for validation.
+    let mut headers_s = BTreeMap::new();
+    for (name, val) in headers.iter() {
+        if let Ok(val_s) = val.to_str() {
+            headers_s.insert(name.as_str().to_string(), val_s.to_string());
+        }
+    }
+
+    // Run our validation on a separate thread, waiting for it's completion.
+    let valid = mz_ore::task::spawn_blocking(
+        || "webhook-validate".to_string(),
+        move || {
+            let mut row = Row::with_capacity(2);
+            let mut packer = row.packer();
+
+            packer.push(Datum::Bytes(&body[..]));
+            packer.push_dict(
+                headers_s
+                    .iter()
+                    .map(|(name, val)| (name.as_str(), Datum::String(val))),
+            );
+
+            // TODO(parkmycar): Is there a way to get a Datum::Map without packing and unpacking a
+            // row?
+            let datums = row.unpack();
+            let body = datums.get(0).copied().unwrap_or(Datum::Null);
```

Comment:
This could be `let body = Datum::Bytes(&body[..]);`?

Reply:
For the symmetry I packed both into a row, but it's probably more performant if I didn't, updated!